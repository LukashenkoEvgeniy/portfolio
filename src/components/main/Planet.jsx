import React from 'react';
import styled from 'styled-components';

export default function Planet({onClick, planet}) {
        return (
            <PlanetWrapper onClick={onClick}>
                <Name>{planet.name}</Name>
                <PlanetImage
                    src={planet.img}
                />
            </PlanetWrapper>
        );
}

const Name = styled.p`
  font-size: 5rem;
  color: aliceblue;
  position: absolute;
  margin: 0;
  z-index: 11;
`;

const PlanetWrapper = styled.div`
  width: 70%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const PlanetDescription = styled.div`
  color: red;
  font-size: 25px;
`;

const PlanetImage = styled.img`
  &:hover {
    transform: scale(1.1)
  }
   transition: all .2s ease-in-out;
   transition-duration: 3s;
   width: 50%;
   @media (max-width: 700px) {
    width: 90%;
  }
  
  z-index: 10;
`;