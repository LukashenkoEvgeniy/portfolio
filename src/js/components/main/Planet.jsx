import React from 'react';
import styled from 'styled-components';

export default class Planet extends React.Component {

    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        return (
            <PlanetWrapper onClick={this.props.onClick}>
                <Name>{this.props.name}</Name>
                <PlanetImage
                    ref={(ref) => this.ground = ref}
                    src={this.props.planetImg}
                />
            </PlanetWrapper>
        );
    }
}

const Name = styled.p`
  font-size: 5rem;
  color: aliceblue;
  position: absolute;
  margin: 0;
`;

const PlanetWrapper = styled.div`
  width: 70%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const PlanetImage = styled.img`
   width: 50%;
   @media (max-width: 700px) {
    width: 90%;
  }
  background-image: url( ${props => props.img});
`;