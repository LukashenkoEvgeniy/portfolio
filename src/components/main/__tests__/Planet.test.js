import Planet from "../Planet";
import React from "react";

describe('Planet test', () => {

    const planet = {
        id: 0,
        name: 'SUN',
        img: null,
        description: 'Сонце — типова зоря головної послідовності спектрального класу G2. Воно майже ідеально сферичне і являє собою гарячу плазму, сплетену магнітними полями. Сонячне випромінювання підтримує життя на Землі (фотони необхідні для початкових стадій процесу фотосинтезу) та визначає клімат нашої планети. Сонце — потужне джерело енергії, яку воно постійно випромінює в усіх ділянках спектра електромагнітних хвиль — від рентгенівських і ультрафіолетових променів до радіохвиль. Це випромінювання сильно впливає на всі тіла Сонячної системи: нагріває їх, позначається на атмосферах планет, дає світло й тепло.'
    };
    const onClick = jest.fn();

    it('Planet on click must called one time', () => {
        const wrapper = shallow(
            <Planet onClick={onClick} planet={planet}/>
        );
        wrapper.simulate('click')
        expect(onClick).toHaveBeenCalledTimes(1)
    });

    it('Planet rendered right name', () => {
        const wrapper = shallow(
            <Planet onClick={onClick} planet={planet}/>
        );
        // console.log(wrapper.debug());
        expect(wrapper.find('Planet__Name').text()).toBe('SUN')
    });

    it('should toMatchSnapshot', () => {
        const wrapper = shallow(
            <Planet planet={planet}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

});