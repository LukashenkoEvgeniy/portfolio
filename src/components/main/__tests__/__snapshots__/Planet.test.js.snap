// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet test should toMatchSnapshot 1`] = `
<Planet__PlanetWrapper>
  <Planet__Name>
    SUN
  </Planet__Name>
  <Planet__PlanetImage
    src={null}
  />
</Planet__PlanetWrapper>
`;
